### PR TITLE
Fix concurrent map access and limit TIME_WAIT connections

### PR DESCRIPTION
**Pull request by mjarco**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -36,10 +43,10 @@ type YamlConfig struct {
 	// List request methods to be logged in synclog in case of backend failure
 	SyncLogMethods []string `yaml:"SyncLogMethods,omitempty"`
 	// Should we keep alive connections with backend servers
```
Comment (ksitak) [inline config/config.go]: comment in wrong line

File: httphandler/httphandler.go [hunk 2]
```diff
@@ -92,17 +87,19 @@ func (h *Handler) validateIncomingRequest(req *http.Request) int {
 // ConfigureHTTPTransport returns http.Transport with customized dialer,
 // MaxIdleConnsPerHost and DisableKeepAlives
 func ConfigureHTTPTransport(conf config.Config) (*http.Transport, error) {
-
-	connDuration, err := time.ParseDuration(conf.ConnectionTimeout)
-	if err != nil {
-		return nil, err
+	maxIdleConnsPerHost := 100
```
Comment (ksitak) [inline httphandler/httphandler.go]: name it defaultMaxIdleConnsPerHost

File: httphandler/httphandler.go [hunk 3]
```diff
@@ -92,17 +87,19 @@ func (h *Handler) validateIncomingRequest(req *http.Request) int {
...
+	if conf.MaxIdleConnsPerHost != 0 {
+		maxIdleConnsPerHost = conf.MaxIdleConnsPerHost
+	}
+	responseHeaderTimeout := 5 * time.Second
```
Comment (ksitak) [inline httphandler/httphandler.go]: name it defaultResponseHeaderTimeout and move up just below defaultMaxIdleConnsPerHost

File: sharding/sharding.go [hunk 4]
```diff
@@ -184,7 +185,14 @@ func NewHandler(conf config.Config) (http.Handler, error) {
...
 	}
-
+	// go func() {
+	// 	for {
+	// 		select {
+	// 		case <- time.After(time.Second):
+	// 			httptransp.CloseIdleConnections()
+	// 		}
+	// 	}
+	// }()
```
Comment (ksitak) [inline sharding/sharding.go]: to remove?

File: config/config.go [hunk 5]
```diff
@@ -34,12 +41,12 @@ type YamlConfig struct {
...
+	SyncLogMethods []string       `yaml:"SyncLogMethods,omitempty"`
+	Client         *ClientConfig  `yaml:"Client,omitempty"`
+	Logging        LoggingConfig  `yaml:"Logging,omitempty"`
+	Metrics        metrics.Config `yaml:"Metrics,omitempty"`
 	// Should we keep alive connections with backend servers
-	Client    *ClientConfig  `yaml:"Client,omitempty"`
-	Logging   LoggingConfig  `yaml:"Logging,omitempty"`
-	Metrics   metrics.Config `yaml:"Metrics,omitempty"`
-	KeepAlive bool           `yaml:"KeepAlive"`
+	DisableKeepAlives bool `yaml:"DisableKeepAlives"`
```
Comment (ksitak) [inline config/config.go]: It's not used anywhere, is it?

Comment (mjarco) [inline config/config.go]: Fixed.

**Pull request merged**